### PR TITLE
Wagmi & Viem Migration

### PR DESCRIPTION
**Pull request by AtelyPham**
## Summary of changes
- Update `chainsConfig` to match the `viem` Chain type.
	- Remove `chain.base`
	- Remove `chain.blockExplorerStub`
- Move all `BigNumber` → `BigInt`
- Update `WebbProvider` to match the new API
- Remove the `logos` library
- Remove `getNativeCurrency` function in the ApiConfg class and fetchOnChainConfig script
- Update the `WebbWeb3Provider` class with the new API
- Update the `Web3RelayerManager` class with the new API
- Update `Web3VAnchorActions` class with the new API
- Update `Update Web3BridgeApi` class with the new API
- Update `Update Web3ChainQuery` class with the new API
- Update `Update Web3WrapUnwrap` class with the new API
- Remove `Web3Provider` class
- Benchmarking max block step on chains
- Store the sidebar state in local storage
- Sync note percentages
- Remove the wasm workers scripts
- Switch account flow on the dapp
- Remove unused dependencies

### Proposed area of change
_Put an `x` in the boxes that apply._

- [x] `apps/bridge-dapp`
- [ ] `apps/hubble-stats`
- [ ] `apps/stats-dapp`
- [x] `apps/webbsite`
- [x] `apps/faucet`
- [ ] `apps/tangle-website`
- [x] `libs/webb-ui-components`

### Reference issue to close (if applicable)
- Closes https://github.com/webb-tools/webb-dapp/issues/1226

### Screen Recording

https://www.loom.com/share/c9344185337c4e69acee5ed736a2e106?sid=f99873b2-ea30-4e81-9e1f-dd3446ba51cf

### CODE REVIEW

File: apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts [hunk 1]
```diff
@@ -175,26 +173,30 @@ export const useMaxFeeInfo = (
...
-      } else if (provider instanceof Web3Provider) {
+      } else if (activeApi instanceof WebbWeb3Provider) {
         const gasAmount = gasLimit[currentTypedChain];
-        const etherProvider = provider.intoEthersProvider();
+        const walletClient = activeApi.walletClient;
+        const publicClient = activeApi.publicClient;
+
+        const { maxFeePerGas, gasPrice, maxPriorityFeePerGas } =
+          await fetchFeeData({
+            chainId: walletClient.chain?.id,
```
Comment (elkremroot) [inline apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts:183]: what if chain.id is undefined?

Comment (elkremroot) [reply apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts:183]: Can you please check for this case.

**Review by elkremroot — CHANGES_REQUESTED** [review]

**Comment by elkremroot** [comment]
Please also fix conflicts

File: apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts (hunk 1 above)
Comment (AtelyPham) [reply apps/bridge-dapp/src/hooks/useMaxFeeInfo.ts:183]: The library manages this, with `chainId` being optional. Also, `fetchFeeData` is a `viem` function. @elkremroot

**Comment by drewstone** [comment]
Lots of new merge conflicts here @AtelyPham

**Comment by drewstone** [comment]
@devpavan04 @elkremroot @vutuanlinh2k2 have you QA'd this PR? Execute the flows in the dApp on the preview sites?

**Comment by elkremroot** [comment]
@drewstone not yet. QA is the last step I do.

**Comment by elkremroot** [comment]
@AtelyPham FYI: 

image

**Comment by drewstone** [comment]
Last commit broke the builds

**Review by elkremroot — APPROVED** [review]

**Pull request merged**